src/arch/x86_64/Mir.zig

```diff
@@ -434,24 +456,36 @@ pub const Inst = struct {
         },
         /// Register, immediate.
         ri: struct {
-            r1: Register,
-            imm: u32,
+            r: Register,
+            i: u32,
         },
         /// Register, followed by custom payload found in extra.
         rx: struct {
-            r1: Register,
+            r: Register,
             payload: u32,
         },
         /// Register with condition code (CC), followed by custom payload found in extra.
         rx_cc: struct {
-            r1: Register,
+            r: Register,
             cc: bits.Condition,
             payload: u32,
         },
-        /// Custom payload followed by an immediate.
-        xi: struct {
+        /// Immediate, followed by Custom payload found in extra.
+        ix: struct {
+            i: u32,
+            payload: u32,
+        },
+        /// Register, register, followed by Custom payload found in extra.
+        rrx: struct {
+            r1: Register,
+            r2: Register,
+            payload: u32,
+        },
+        /// Register, immediate, followed by Custom payload found in extra.
+        rix: struct {
```

Comment:
Let's try and keep the data here at 8 bytes max. So instead of creating `rix`, why not put the immediate and payload in extra data, and use `rx` payload instead?

Reply:
I will add an assert so that we know in the future when we exceed this 8 bytes.  I could have guessed the current value by reading the whole struct, but that's not something I want to have to do manually in the middle of adding features.

Reply:
TIL there even is an `mri` encoding!